src/js/sources.js

```diff
@@ -191,34 +167,57 @@ export function ytcSource(window) {
     runQueue();
   };
 
-  const getTimestamp = (data, message) => data.isReplay
-    ? message.showtime
-    : (Date.now() + message.showtime) / 1000;
-
-  const extractToMsg = data => message => ({
-    timestamp: getTimestamp(data, message), message
+  const extractToMsg = () => (message) => ({
+    timestamp: message.showtime / 1000,
+    message
   });
 
-  const pushMessagesToQueue = data => data.messages
+  const pushMessagesToQueue = (messages) => messages
     .sort(lessMsg)
-    .map(extractToMsg(data))
+    .map(extractToMsg(messages))
     .forEach(item => queued.push(item));
 
-  window.addEventListener('message', d => {
-    const data = getYTCData(d);
-    updateVideoProgressBeforeMessages(data);
-    
-    if (data.type === 'messageChunk') {
+  /** Connect to background messaging as client */
+  const port = chrome.runtime.connect();
+  let portRegistered = false;
+  const registerClient = (frameInfo) => {
+    port.postMessage({
+      type: 'registerClient',
+      frameInfo: frameInfo,
+      getInitialData: false
+    });
+    portRegistered = true;
+  };
+
+  const params = new URLSearchParams(window.location.search);
+  const isPopout = params.get('popout');
```

Comment:
Please move ``isPopout`` and ``params`` to ``src/js/constants.js``

Reply:
Should I move all other `new URLSearchParams(window.location.search)` and `params.get()` usages to constants as well? They're used in a bunch of other places as well.

Reply:
Yea, that would make the code a lot better.